### PR TITLE
Added a JSONEncoder subclass that permits the Calendar tree to be encoded as JSON

### PR DESCRIPTION
**Pull request by rbpasker**
here's a working example

``` python

from json import JSONEncoder
import json
from datetime import datetime
from icalendar import Calendar, vDDDTypes, Event, ICalendarEncoder
import pprint

cal = Calendar.from_ical(open('test.ics','rb').read())

for event in cal.walk(name="VEVENT"):
    suspect = json.dumps(event, cls=ICalendarEncoder)
    working = json.loads(suspect)

pprint.pprint(working)

```

### CODE REVIEW

File: src/icalendar/prop.py [hunk 1]
```diff
@@ -1553,3 +1556,19 @@ def from_ical(self, name, value):
...
+        try:
+            if obj.__module__.startswith("icalendar.prop"):
+                return (obj.to_ical())
+        except AttributeError:
+            pass
+
+        if isinstance(obj, datetime):
+            return (obj.now().strftime('%Y-%m-%dT%H:%M:%S'))
+        
+        return JSONEncoder.default(self,obj)
```
Comment (thet) [inline src/icalendar/prop.py:1573]: wouldn't a call to super be more appropriate here?

File: src/icalendar/prop.py [hunk 2]
```diff
@@ -1553,3 +1556,19 @@ def from_ical(self, name, value):
         type_class = self.for_property(name)
         decoded = type_class.from_ical(value)
         return decoded
+
+
+class ICalendarEncoder(JSONEncoder):
+    def default(self, obj, markers=None):
+
```
Comment (thet) [inline src/icalendar/prop.py:1563]: markers is not used

File: src/icalendar/prop.py [hunk 3]
```diff
@@ -1553,3 +1556,19 @@ def from_ical(self, name, value):
         type_class = self.for_property(name)
         decoded = type_class.from_ical(value)
         return decoded
+
+
+class ICalendarEncoder(JSONEncoder):
```
Comment (thet) [inline src/icalendar/prop.py:1561]: A different name would be better. Interfaces are always named ISomething  by convention. ICalendarEncoder looks like an interface.

Comment (rbpasker) [inline src/icalendar/prop.py:1561]: Sure, go ahead and fix it

File: src/icalendar/prop.py (hunk 1 above)
Comment (rbpasker) [inline src/icalendar/prop.py:1573]: sure, go ahead and fix it

**Comment by untitaker** [comment]
Closing due to inactivity, if there's continued interest in fixing up this PR please open up a new one!

**Pull request closed without merging**